connect/runtime/src/main/java/org/apache/kafka/connect/cli/ConnectDistributed.java

```diff
@@ -125,7 +123,7 @@ public Connect startConnect(Map<String, String> workerProps) {
         try {
             connect.start();
             // herder has initialized now, and ready to be used by the RestServer.
-            provider.setHerder(herder);
+            rest.start(herder, plugins);
```

Comment:
since a lot of these changes are effectively removing HerderProvider and moving the codebase back to how it functioned previously, we might be able to push this `rest.start(..)` into `org.apache.kafka.connect.runtime.Connect#start()`. WDYT?